crates/broker/src/order_picker.rs

```diff
@@ -704,42 +706,64 @@ where
 {
     type Error = OrderPickerErr;
     fn spawn(&self) -> RetryRes<Self::Error> {
-        let picker_copy = self.clone();
+        let picker = self.clone();
 
         Box::pin(async move {
             tracing::info!("Starting order picking monitor");
-            let pricing_semaphore = Arc::new(Semaphore::new(MAX_PRICING_BATCH_SIZE as usize));
 
-            // This should be the only task holding the lock, so it is just locked while the service
-            // is running.
-            let mut rx = picker_copy.new_order_rx.lock().await;
+            let read_capacity = || -> Result<usize, Self::Error> {
+                let cfg = picker.config.lock_all().map_err(|err| {
+                    OrderPickerErr::UnexpectedErr(anyhow::anyhow!("Failed to read config: {err}"))
+                })?;
+                Ok(cfg.market.max_pricing_batch_size as usize)
+            };
+
+            let mut current_capacity = read_capacity().map_err(SupervisorErr::Fault)?;
+            let mut tasks: JoinSet<()> = JoinSet::new();
+            let mut rx = picker.new_order_rx.lock().await;
+            let mut capacity_check_interval = tokio::time::interval(MIN_CAPACITY_CHECK_INTERVAL);
+            let mut pending_orders: VecDeque<Box<OrderRequest>> = VecDeque::new();
 
             loop {
-                // Wait for a new order from the channel - lock the mutex only when receiving
-                let order = rx.recv().await.ok_or_else(|| {
-                    // This should be unrecoverable if the sender is dropped.
-                    SupervisorErr::Fault(OrderPickerErr::UnexpectedErr(anyhow::anyhow!(
-                        "Order channel closed unexpectedly"
-                    )))
-                })?;
+                tokio::select! {
+                    // This channel is cancellation safe, so it's fine to use in the select!
+                    order = rx.recv() => {
+                        let order = order.ok_or_else(|| {
+                            SupervisorErr::Fault(OrderPickerErr::UnexpectedErr(anyhow::anyhow!(
+                                "Order channel closed unexpectedly"
+                            )))
+                        })?;
+
+                        pending_orders.push_back(order);
```

Comment:
I'd like to be able to understand e2e whats happening with each order with 1 log query, so lets log the order id when its enqueued and dequeued